src/java.base/share/classes/java/io/ObjectStreamClass.java

```diff
@@ -2231,13 +2168,8 @@ void getObjFieldValues(Object obj, Object[] vals) {
              */
             for (int i = numPrimFields; i < fields.length; i++) {
                 switch (typeCodes[i]) {
-                    case 'L':
-                    case '[':
-                        vals[offsets[i]] = unsafe.getReference(obj, readKeys[i]);
-                        break;
-
-                    default:
-                        throw new InternalError();
+                    case 'L', '[' -> vals[offsets[i]] = unsafe.getReference(obj, readKeys[i]);
+                    default       -> throw new InternalError();
```

Comment:
suggest:

    vals[offsets[i]] = switch (typeCodes[i]) {
        case 'L', '[' -> unsafe.getReference(obj, readKeys[i]);
        default       -> throw new InternalError();

Reply:
Comment addressed in e503ed2